lib/dry/validation/evaluator.rb

```diff
@@ -147,6 +147,22 @@ def value
         values[key_name]
       end
 
+      # Return if the value under the default key is available
+      #
+      # This is useful when dealing with rules for optional keys
+      #
+      # @example
+      #   rule(:age) do
+      #     key.failure(:invalid) if value < 18 if value? && value < 18
+      #   end
+      #
+      # @return [Boolean]
+      #
+      # @api public
+      def value?
```

Comment:
I wonder if this is actually the right name? In my mind `key?` actually seems like the more appropriate name? To my instincts, at least, `value?` feels like something that's returning the truthiness of the value...

Reply:
If you consider it in usage, I think it feels pretty good named as `#key?` too...

```ruby
      rule(:login) do
         if key?
           key.failure('too short') if value.length < 3
         end
       end
```